polkadot/node/network/protocol/src/grid_topology.rs

```diff
@@ -111,6 +119,11 @@ impl SessionGridTopology {
 
 		Some(grid_subset)
 	}
+
+	/// Tells if a given peer id is validator in a session
+	pub fn is_validator(&self, peer: &PeerId) -> bool {
```

Comment:
[ExtendedSessionInfo](https://github.com/paritytech/polkadot-sdk/blob/7cbe0c76ef8fd2aabf9f07de0156941ce3ed44b0/polkadot/node/subsystem-util/src/runtime/mod.rs#L92) already tells you whether you are validator in the current session.